### PR TITLE
Add startup and liveness probes based on dogu.json "Healthchecks" entries

### LINKED ISSUE #15: Add liveness/readiness/startup probes
Add liveness/readiness/startup probes on dogu installation based on the HealthChecks entries in the dogu.json

### PR DESCRIPTION
**Pull request by robertauer**
### Added
- Add startup probe based on state at dogu deployment generation; #15
- Add liveness probe based on tcp port at dogu deployment generation; #15

Resolves #15

### CODE REVIEW

File: controllers/resource/resource_generator_test.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package resource_test
 
 import (
+	_ "embed"
```
Comment (ppxl) [inline controllers/resource/resource_generator_test.go:4]: :+1:

**Review by ppxl — CHANGES_REQUESTED** [review]
This looks like a nice and lean solution! Maybe we should look into visibility of structs and methods, though.

File: controllers/dependency/dogu_dependency_validator.go [hunk 2]
```diff
@@ -14,6 +14,11 @@ type ErrorDependencyValidation struct {
 	Dependency  core.Dependency
 }
 
+// DoguDependencyChecker is used to  check a single dependency of a dogu
+type DoguDependencyChecker interface {
```
[the remark this answers is not in the document]
Comment (robertauer) [reply controllers/dependency/dogu_dependency_validator.go:18]: @nhinze23

File: controllers/dependency/dogu_dependency_validator.go [hunk 3]
```diff
@@ -31,7 +36,7 @@ func (e *ErrorDependencyValidation) Requeue() bool {
 
 // DoguDependencyValidator is responsible to check if all dogu dependencies are valid for a given dogu
 type DoguDependencyValidator struct {
-	DoguDependencyChecker *dependencies.DoguDependencyChecker
+	DoguDependencyChecker DoguDependencyChecker
 }
```
[the remark this answers is not in the document]
Comment (robertauer) [reply controllers/dependency/dogu_dependency_validator.go:40]: @nhinze23

File: controllers/resource/resource_generator_test.go (hunk 1 above)
Comment (robertauer) [reply controllers/resource/resource_generator_test.go:4]: All credits go to my IDE :D

**Comment by ppxl** [comment]
LGTM

**Pull request merged**